demisto_sdk/commands/split/ymlsplitter.py

```diff
@@ -195,7 +200,7 @@ def extract_code(
         else:
             lang_type = self.yml_data["type"]
         ext = TYPE_TO_EXTENSION[lang_type]
-        code_file_path = code_file_path.with_suffix(ext)
+        code_file_path = code_file_path.parent / f"{code_file_path.name}{ext}"
```

Comment:
This fixes an issue where files with `.` in their name, for example `Test v2.1`, would have `.1` set as their suffix by pathlib, which is then replaced by `.py` causing `Test v2.1.py` to be named `Test v2.py`.

To reproduce, run:
```python
str(Path("path/to/file/Test 2.1").with_suffix(".py"))
```

Reply:
1. separate PR
2. can `rstrip` help?

Reply:
[Opened a separate PR](https://github.com/demisto/demisto-sdk/pull/3698)